src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToDefenceSpecCallbackHandler.java

```diff
@@ -704,6 +705,20 @@ private CallbackResponse buildJudgmentAmountSummaryDetails(CallbackParams callba
             .build();
     }
 
+
+    private void checkPartyAddress(CaseData caseData) {
+
+        if (null != caseData.getApplicant1()
+            && null == caseData.getApplicant1().getPrimaryAddress()
+            && null != caseData.getSpecApplicantCorrespondenceAddressdetails()) {
+            caseData.getApplicant1().setPrimaryAddress(caseData.getSpecApplicantCorrespondenceAddressdetails());
+        }
+        if (null != caseData.getRespondent1()
+            && null == caseData.getRespondent1().getPrimaryAddress()
+            && null != caseData.getRespondent1DetailsForClaimDetailsTab()) {
+            caseData.getRespondent1().setPrimaryAddress(caseData.getRespondent1DetailsForClaimDetailsTab().getPrimaryAddress());
+        }
+
```

Comment:
A few things I'd like to ask!
1. Shouldn't this involve applicant 2 and respondent 2 as well? 
2. why is it for applicant we take a spec address but for respondent is from the claim details tab? 

May i suggest that we use caseData before?
```
CaseData oldCaseData = caseDetailsConverter.toCaseData(callbackParams.getRequest().getCaseDetailsBefore());
oldCaseData.getRespondent1().getPrimaryAddress()
```
The above code just returns the data from before the event is run. So you should be able to just get the data from there.

Reply:
done